pages/in/register/index.vue

```diff
@@ -41,56 +123,274 @@ export default {
+      if (this.currentTab === TAB_OPTIONS.ERROR) {
+        return this.errorTitle;
+      }
+      if (this.getWalletConnectURI) {
+        return this.$t('V2_WalletConnectQRCodeModal_Title');
+      }
+      return this.$t('DialogV2.title.signIn');
+    },
+    tabOptions() {
+      return TAB_OPTIONS;
     },
   },
-  created() {
-    this.setAuthDialog({ isShow: true });
+  watch: {
+    isRedirectSignIn: {
+      immediate: true,
+      async handler(isRedirectSignIn) {
+        if (isRedirectSignIn) {
+          this.currentTab = TAB_OPTIONS.REDIRECT;
+          await this.handleAuthSignIn();
+        }
+      },
+    },
   },
-  mounted() {
-    const { redirect_sign_in: isRedirectSignIn } = this.$route.query;
-    if (this.getUserIsRegistered) {
-      logTrackerEvent(this, 'RegFlow', 'AlreadyRegistered', 'AlreadyRegistered', 1);
-      const router = this.$router;
-      const route = this.$route;
-      if (!isRedirectSignIn && !tryPostLoginRedirect({ route })) {
-        this.doPostAuthRedirect({ router, route });
-      }
-    } else {
-      window.addEventListener(this.unloadEventName, this.logPageUnload, false);
-      this.logPageload();
+  async mounted() {
+    if (this.currentTab === TAB_OPTIONS.LOGIN && !this.isRedirectSignIn) {
+      await this.handleConnectWallet();
     }
   },
-  beforeDestroy() {
-    this.setAuthDialog({ isShow: false });
-    window.removeEventListener(this.unloadEventName, this.logPageUnload, false);
-  },
   methods: {
     ...mapActions([
-      'setAuthDialog',
+      'handleConnectorRedirect',
+      'loginUser',
       'doPostAuthRedirect',
+      'newUser',
+      'updateUserAvatar',
     ]),
-    logPageUnload() {
-      let value = 1;
-      if (window.performance) {
-        value = Math.round(performance.now());
+    async onClickLoginButton() {
+      await this.handleConnectWallet();
+    },
+    onClickDialogCloseButton() {
+      this.currentTab = TAB_OPTIONS.LOGIN;
+      this.errorCode = '';
+      this.error = '';
+      this.signInPayload = {};
+      this.authcoreUserData = {};
+      this.handleConnectWallet();
+    },
+    setError(code, error) {
+      this.currentTab = TAB_OPTIONS.ERROR;
+      this.errorCode = code;
+      this.error = error;
+    },
+
+
+    // Login
+    async handleConnectWallet() {
+      try {
+        await this.connectWallet();
+        const successLogin = await this.login();
+        if (successLogin) {
+          this.redirectAfterSignIn();
+        }
+      } catch (error) {
+      // eslint-disable-next-line no-console
+        this.setError(error.message, error);
+      }
+    },
+    async handleAuthSignIn() {
+      const { method, code } = this.$route.query;
+      if (method && code) {
+        try {
+          const user = await this.handleConnectorRedirect({
+            method,
+            params: { code },
+          });
+          this.authcoreUserData = user;
+          const successLogin = await this.login({ idToken: user.idToken });
```

Comment:
We should check if `user.idToken` exists here

Reply:
[♻️ Improve code readability based on comment suggestions](https://github.com/likecoin/like-co/pull/1740/commits/de152633a72f64a26f8041d517b6aef6cbddc2d9)